app/controllers/course_summaries_controller.rb

```diff
@@ -15,4 +15,118 @@ def populate
   def get_marking_scheme_details
     redirect_to url_for(controller: 'marking_schemes', action: 'populate')
   end
+
+  def download_csv_grades_report
+    assignments = Assignment.order(:id)
+    csv_string = CSV.generate do |csv|
+      # Populates the entire CSV with headers, students and their marks
+      populated_csv = populate_students_and_marks(csv)
+    end
+    course_name = "#{COURSE_NAME}"
+    course_name_underscore = course_name.squish.downcase.tr(" ", "_")
+    send_data csv_string, disposition: 'attachment',
+                         filename: "#{course_name_underscore}_grades_report.csv"
+  end
+end
+
+# ---- HELPER FUNCTIONS FOR DOWNLOAD_CSV_GRADES_REPORT BELOW-----#
+
+# Populates the headers in the CSV
+def populate_header_titles(assignments)
+  header = ['Username']
+  assignments.each do |assignment|
+    header.push(assignment.short_identifier)
+  end
+  header.push('Total marks')
+end
+
+# Checks grouping, submission and it's mark for nil.
+def fill_row_with_marks(grouping, row)
+  if check_for_nil_fields(grouping)
+    row.push('')
+  else
+    row.push(grouping.current_submission_used.get_latest_result.total_mark /
+                                       grouping.assignment.total_mark * 100)
+  end
+  row
+end
+
+# Loads up the CSV with headers, students and marks.
+def populate_students_and_marks(csv)
```

Comment:
Assignment Branch Condition size for populate_students_and_marks is too high. [16.12/15]